sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -55,6 +55,7 @@ statement
         rowFormat?  createFileFormat? locationSpec?
         (TBLPROPERTIES tablePropertyList)?
         (AS? query)?                                                   #createTable
+    | createTableHeader LIKE source=tableIdentifier                    #createTableLike
```

Comment:
Not sure if we want to support `CREATE TABLE IF NOT EXIST x LIKE y`? It is not on the Hive manual, but `HiveQuerySuite` has tests using this syntax...